tracer/test/test-applications/integrations/Samples.AWS.Kinesis/Program.cs

```diff
@@ -0,0 +1,44 @@
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Amazon.Runtime;
+using Amazon.Kinesis;
+
+namespace Samples.AWS.Kinesis
+{
+    public class Program
+    {
+        private static async Task Main(string[] args)
+        {
+            // Set up AmazonSQSConfig and redirect to the local message queue instance
+            var kinesisClient = GetAmazonKinesisClient();
+
+#if NETFRAMEWORK
+            SyncHelpers.SendKinesisTasks(kinesisClient);
+#endif
+            await AsyncHelpers.StartKinesisTasks(kinesisClient);
```

Comment:
FYI running this locally on .NET Framework, when the AsyncHelpers.StartKinesisTasks runs immediately after the SyncHelpers.StartKinesisTasks, the CreateStreamAsync throws an exception that the stream already exists. This looks like it happens because the `DeleteStream` call returns but it doesn't mean that the stream is deleted immediately, it may still be in the `DELETING` state.

To workaround this, I have two ideas:
1) Create a `SyncStream` and a `AsyncStream` so that we don't have to wait on the stream deletion
2) Create a loop to check that the stream is closed.

The second option might be the most reliable so that if this test runs consecutively we can always ensure that when an application exits, then we know the stream was deleted. In fact, we could actually combine solutions 1 and 2! What do you think?

Reply:
In a87def01baf7fd2c7dfd02c43213b9203138ce0d I went with an option 3, which was to introduce a delay at the end of each of the `StartKinesisTasks` methods. This should result in the stream being deleted before running the Async routine or before the next application is started